nifi-frontend/src/main/frontend/apps/nifi/src/app/pages/flow-designer/ui/canvas/items/processor/edit-processor/edit-processor.component.html

```diff
@@ -290,19 +308,79 @@ <h2 mat-dialog-title>
         </mat-tab>
     </mat-tab-group>
     @if ({ value: (saving$ | async)! }; as saving) {
-        <mat-dialog-actions align="end">
-            @if (readonly) {
-                <button mat-flat-button mat-dialog-close>Close</button>
-            } @else {
-                <button mat-button mat-dialog-close>Cancel</button>
-                <button
-                    [disabled]="!editProcessorForm.dirty || editProcessorForm.invalid || saving.value"
-                    type="button"
-                    (click)="submitForm()"
-                    mat-flat-button>
-                    <span *nifiSpinner="saving.value">Apply</span>
-                </button>
-            }
+        <mat-dialog-actions align="start">
+            <div class="flex w-full justify-between items-center">
+                <div>
+                    @if (isStoppable()) {
+                        <button type="button" mat-stroked-button [matMenuTriggerFor]="operateMenu">
+                            <div class="flex items-center">
+                                <i class="mr-2 success-color-default fa fa-play"></i>Running<i
+                                    class="ml-2 -mt-1 fa fa-sort-desc"></i>
+                            </div>
+                        </button>
+                    } @else if (isRunnable()) {
+                        <button type="button" mat-stroked-button [matMenuTriggerFor]="operateMenu">
+                            <div class="flex items-center">
+                                <i class="mr-2 error-color-variant fa fa-stop"></i>Stopped<i
+                                    class="ml-2 -mt-1 fa fa-sort-desc"></i>
+                            </div>
+                        </button>
+                    } @else {
+                        <button
+                            type="button"
+                            mat-stroked-button
+                            [disabled]="isStopping()"
+                            [matMenuTriggerFor]="operateMenu">
+                            <div class="flex items-center">
+                                @if (isInvalid()) {
+                                    <i class="mr-2 fa fa-warning caution-color"></i>
+                                } @else if (isDisabled()) {
+                                    <i class="mr-2 icon icon-enable-false"></i>
+                                } @else {
+                                    <i class="mr-2 fa fa-circle-o-notch fa-spin primary-color"></i>
+                                }
```

Comment:
If the user cannot operate the Processor, this will result in the showing the spinner.

<img width="156" alt="Screenshot 2024-12-09 at 3 29 32 PM" src="https://github.com/user-attachments/assets/138a2d55-b681-432e-81d6-62c11baf75de">

I think in this case, the button should be disabled since they don't have permission to operate this Processor.

Reply:
Really nice catch! Updated to reflect this additional consideration.